From b024977502b36f32fa98f1d1b85a161460fe750c Mon Sep 17 00:00:00 2001
From: CI Assistant <yossi.ilkanaev@intel.com>
Date: Thu, 23 Aug 2018 11:14:52 +0200
Subject: [PATCH] Merge pull request #138 in SW_UGW/linux from
 bugfix/DRVLIB_SW-862-falcon-mountain-lan-port-1-only-works-when-phy0-and-pyh1-activated
 to master

* commit 'a6b08f6f970ab5e131f0d12648895cbea941b934':
  Fix issue: default aggregate mode is invalid if DT is disabled
---
 drivers/phy/intel/phy-intel-combo.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/phy/intel/phy-intel-combo.c b/drivers/phy/intel/phy-intel-combo.c
index 648e03fc1822..9041633475e8 100644
--- a/drivers/phy/intel/phy-intel-combo.c
+++ b/drivers/phy/intel/phy-intel-combo.c
@@ -4,7 +4,6 @@
  *
  * Copyright (C) 2017 Intel Corporation.
  */
-
 #include <linux/kernel.h>
 #include <linux/module.h>
 #include <linux/init.h>
@@ -72,7 +71,7 @@ enum {
 };
 
 enum aggregated_mode {
-	PHY_SL_MODE = 1, /* Single Lane mode */
+	PHY_SL_MODE = 0, /* Single Lane mode */
 	PHY_DL_MODE,	/* Dual Lane mode */
 };
 
@@ -773,6 +772,9 @@ static int intel_phy_dt_parse(struct intel_combo_phy *priv,
 	iphy->enable = of_device_is_available(np);
 	priv->enable[idx] = iphy->enable;
 
+	dev_dbg(dev, "PHY(%u:%u) mode: %u, role: %u, enable %u\n",
+		COMBO_PHY_ID(iphy), PHY_ID(iphy),
+		iphy->phy_mode, iphy->phy_role, iphy->enable);
 	return 0;
 }
 
@@ -902,6 +904,7 @@ static int intel_combo_phy_set_mode(struct intel_combo_phy *priv)
 	if (priv->cb_phy_mode == COMBO_PHY_MODE_MAX)
 		return -EINVAL;
 
+	dev_dbg(priv->dev, "combo mode: %u\n", priv->cb_phy_mode);
 	if (priv->soc_data->combo_phy_mode_set)
 		priv->soc_data->combo_phy_mode_set(priv);
 
